source/manual/howto-ssh-to-machines-in-aws.html.md

```diff
@@ -4,81 +4,132 @@ title: SSH into AWS machines
 
-Ensure you have cloned the `govuk-aws` repository:
+## Setup
 
-```
+Clone the `govuk-aws` repository and symlink to make `govukcli` executable globally:
+
+```sh
 cd ~/govuk
 git clone https://github.com/alphagov/govuk-aws
-```
-
-Add a symlink to ensure you use the current version:
-
-```
 ln -s ~/govuk/govuk-aws/tools/govukcli /usr/local/bin/govukcli
 ```
 
-## Connecting to instances
+You will also need to have followed the [Get SSH access to integration](/manual/get-ssh-access.html)
+instructions.
 
-The tool uses the idea of "contexts", where a "context" is a specific environment.
+## Usage
 
-To view the contexts run:
+### Local dev machine
+
+To view all possible contexts, run:
 
 `govukcli list-contexts`
 
-Set a context to interact with that environment:
+To set a persistent context:
 
 `govukcli set-context integration`
 
-### AWS
+Now when you SSH into an instance, it will be to one running on the specified environment:
 
-In AWS you must specify a node class only. You must use `_` rather than `-` when specifying
-the class.
+```sh
+# SSH into a 'calculators_frontend' node on Integration, as per the context you set
+govukcli ssh calculators_frontend
+```
 
-`govukcli ssh calculators_frontend`
+Alternatively, you can set a context for the current command only by passing a CLI parameter:
 
-This will take you to a random instance of that class.
+```sh
+# SSH into a 'calculators_frontend' node on Staging, ignoing the context you set earlier
+govukcli ssh --context staging calculators_frontend
+```
 
-If you wish to go to a specific node, you can log into the jumpbox:
+Note that the SSH examples above will SSH you into a random machine of the right node class.
+You can find out which class you need by finding the corresponding [app page](https://docs.publishing.service.gov.uk/apps.html).
+`govukcli` finds a matching machine via the 'jumpbox', which you can also SSH into directly:
 
 `govukcli ssh jumpbox`
 
-Return a list of available instances:
+#### AWS vs Carrenza
+
+In Carrenza, you can SSH into a specific node:
+
+```sh
+# SSH into the `backend-1.backend.integration` node
+govukcli ssh backend-1
+```
+
+...or you can SSH into a random machine of the right node class:
+
+```sh
+# This may SSH into `backend-1.backend.integration`, or into a different node
+govukcli ssh backend
+```
+
+In AWS, you can only SSH by node class.
+
+### Jumpbox
+
+The jumpbox is a special node that knows about all of the other nodes in its environment.
+
+List the IP addresses of every node in the environment:
+
+`govuk_node_list`
+
+This long list of IPs is not very useful on its own, but you can filter it by node class:
+
+`govuk_node_list -c backend`
+
+And if you can't remember the names of the node classes, there's a built-in helper:
+
+`govuk_node_list --classes`
+
+Once you have found the IP of the machine you want to SSH into, you can manually SSH
+directly from the jumpbox machine:
+
+`ssh ip-10-1-5-22.eu-west-1.compute.internal`
```

Comment:
You can also do this from your local machine by tacking the environment to the end of the address. 

e.g. `ssh ip-10-1-5-22.eu-west-1.compute.internal.integration`

Reply:
Didn't know that, have added a line 👍 thanks